xla/service/gpu/gemm_rewriter.cc

```diff
@@ -182,7 +182,7 @@ bool IsSupportedF8Pattern(HloInstruction *instr, HloInstruction *&x,
                m::AnyOf<HloInstruction>(
                    m::Bitcast().WithPredicate(preserves_element_type),
                    m::Broadcast(), m::Copy(), m::Pad(), m::Reshape(),
-                   m::Slice()))) {
+                   m::Slice(), m::AllGather()))) {
```

Comment:
This effectively causes an FP16 or FP32 all-gather to be converted to an FP8 all-gather. Are FP8 all-gathers supported? 

Also why are all-gathers being introduced between the dequantize and the matmul?

Reply:
Yes. When both A and B are sharded in both column and row dimensions, all-gather is inserted ahead of matmul.